From d3bc82770474761571f950347560d62a35d519d7 Mon Sep 17 00:00:00 2001
From: Chun-Wei Chen <jacky82226@gmail.com>
Date: Mon, 16 Nov 2020 22:06:17 -0800
Subject: [PATCH] Resolve SetTotalBytesLimit warning: only take one parameter
 for protobuf>=3.11 (#3112)

* resolve proto setTotal warning

Signed-off-by: Chun-Wei Chen <jacky82226@gmail.com>

* correct 3110000 to 3011000 for proto version

Signed-off-by: Chun-Wei Chen <jacky82226@gmail.com>
---
 third_party/onnx/onnx/proto_utils.h | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/third_party/onnx/onnx/proto_utils.h b/third_party/onnx/onnx/proto_utils.h
index ec125fde1b..284c1059d8 100644
--- a/third_party/onnx/onnx/proto_utils.h
+++ b/third_party/onnx/onnx/proto_utils.h
@@ -27,11 +27,17 @@ inline std::string ProtoDebugString(const Message& proto) {
 
 template <typename Proto>
 bool ParseProtoFromBytes(Proto* proto, const char* buffer, size_t length) {
-  // Total bytes hard limit / warning limit are set to 1GB and 512MB
-  // respectively.
   ::google::protobuf::io::ArrayInputStream input_stream(buffer, static_cast<int>(length));
   ::google::protobuf::io::CodedInputStream coded_stream(&input_stream);
-  coded_stream.SetTotalBytesLimit((2048LL << 20) - 1, 512LL << 20);
+  int total_bytes_limit = (2048LL << 20) - 1;
+#if GOOGLE_PROTOBUF_VERSION >= 3011000
+    // Only take one parameter since protobuf 3.11
+    coded_stream.SetTotalBytesLimit(total_bytes_limit);
+#else
+    // Total bytes hard limit / warning limit are set to 2GB and 512MB respectively.
+    coded_stream.SetTotalBytesLimit(total_bytes_limit, 512LL << 20);
+#endif
+
   return proto->ParseFromCodedStream(&coded_stream);
 }
 
